src/main/java/seedu/address/logic/commands/UnmarkCommand.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.task.Task;
+
+/**
+ * Unmark (label as incomplete) a task identified using it's displayed index from the task list.
+ */
+public class UnmarkCommand extends Command {
+
+    public static final String COMMAND_WORD = "unmark";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+        + ": Unmark the task identified by the index number used in the displayed task list.\n"
+        + "Parameters: INDEX (must be a positive integer)\n"
+        + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_UNMARK_TASK_SUCCESS = "Unmarked Task: %1$s";
+
+    private final Index targetIndex;
+
+    public UnmarkCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Task> lastShownList = model.getFilteredTaskList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        Task taskToUnmark = lastShownList.get(targetIndex.getZeroBased());
+        Task unmarkedTask = taskToUnmark.unmark();
+        model.setTask(taskToUnmark, unmarkedTask);
+
+        return new CommandResult(String.format(MESSAGE_UNMARK_TASK_SUCCESS, taskToUnmark));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof UnmarkCommand // instanceof handles nulls
+            && targetIndex.equals(((UnmarkCommand) other).targetIndex)); // state check
+    }
+}
```

Comment:
Would it better to return a different message eg "Task is already marked incomplete" instead of "unmarked task" when the user tries to unmark a task that is already unmarked? Same thing for mark.